src/main/java/com/guysfromusa/carsgame/control/GameEngine.java

```diff
@@ -29,30 +28,36 @@
 
     private final CarService carService;
 
+    private final CarMoveHandler carMoveHandler;
+
     @Inject
     public GameEngine(ActiveGamesContainer activeGamesContainer,
                       ApplicationEventPublisher applicationEventPublisher,
-                      CarService carService) {
+                      CarService carService,
+                      CarMoveHandler carMoveHandler) {
         this.activeGamesContainer = notNull(activeGamesContainer);
         this.applicationEventPublisher = notNull(applicationEventPublisher);
-        this.carService = carService;
+        this.carService = notNull(carService);
+        this.carMoveHandler = notNull(carMoveHandler);
     }
 
     @Async
-    public void handleMoves(List<Command> commands, String gameId) {
-        GameState gameState = activeGamesContainer.getGameState(gameId);
+    public void handleMoves(List<Command> commands, String gameName) {
+        GameState gameState = activeGamesContainer.getGameState(gameName);
 
-        //TODO for all commands calculate movements and collisions
+        commands.stream()
+                .map(command -> (MoveCommand)command)
+                .map(moveCmd -> Tuple.of(moveCmd.getFuture(), new MoveData(gameState, moveCmd)))
+                .forEach(moveData -> {
+                    //TODO catch errors and completeExceptionally
+                    carMoveHandler.handleMoveCommand(moveData._2);
+                    moveData._1.complete(gameState.getAllCars());
+                });
 
-        //TODO store all state in DB
 
-        //TODO update memory state
+        //TODO error handling
 
-        commands.forEach(message -> {
-            CompletableFuture<String> future = message.getFuture();
-            log.info("Complete message");
-            future.complete("{status:'OK'}");
-        });
+        //TODO store all state in DB
```

Comment:
we should store data in DB here in one transaction not in few transactions during the process